Initial description of a Lattice-Boltzmann chapter

mdavezac opened the pull request:
The code on which we would like to base this is added as a submodule.
The idea would be to meld code directly from the repo and text via dexy.
Is that OK? Or is a different solution needed?

Commit:
Adding start of lattice boltzmann

A git submodule is added that refers to the original code. The idea would be to
incorporate in the text directly via [dexy](http://dexy.it/), to show actual
testing rather than toy problem stuff. We can always remove it later on.

gvwilson commented:
I'd like to avoid submodules - our experience with them has been that 
they break or misbehave in ways that are difficult for non-expert git 
users to diagnose and correct.

Commit:
removing submodule

Commit:
Revert "removing submodule"

This reverts commit 46ee45607eb3d363f761aab6c5f40abdc45a3877.

Commit:
Submodules are difficult to deal with. Point well made.

Removing submodule the right way around.

Commit:
Adding index.md file for easy serve of LB stuff

The LB stuff should be available as http://{base}/lattice-boltzmann, where
{base} could be the github page location, the localhost, or something ...
